commit b2343fd02c4fff94957f0151b73daa0a1f7fec49
Author: Ethan A Merritt <merritt@u.washington.edu>
Date:   Tue Jan 14 21:54:14 2025 -0800

    canvas:  handle nonlinear x2 or y2 axis with an incomplete definition
    
    Actually "handle" means "ignore".
    But now it doesn't segfault trying to probe a non-existant link function.
    
    Bug 2755

diff --git term/canvas.trm term/canvas.trm
index a919b09bd..bfe567832 100644
--- term/canvas.trm
+++ term/canvas.trm
@@ -646,7 +646,7 @@ CANVAS_text()
 	} else
 	    fprintf(gpoutfile, "gnuplot.plot_axis_x2min = \"none\"\n");
 	if (axis_array[SECOND_X_AXIS].linked_to_primary
-	&&  axis_array[FIRST_X_AXIS].link_udf->at) {
+	&&  axis_array[FIRST_X_AXIS].link_udf && axis_array[FIRST_X_AXIS].link_udf->at) {
 	    fprintf(gpoutfile, "gnuplot.x2_mapping = function(x) { return x; };");
 	    fprintf(gpoutfile, "  // replace returned value with %s\n",
 			axis_array[FIRST_X_AXIS].link_udf->definition);
@@ -657,7 +657,7 @@ CANVAS_text()
 	} else
 	    fprintf(gpoutfile, "gnuplot.plot_axis_y2min = \"none\"\n");
 	if (axis_array[SECOND_Y_AXIS].linked_to_primary
-	&&  axis_array[FIRST_Y_AXIS].link_udf->at) {
+	&&  axis_array[FIRST_Y_AXIS].link_udf && axis_array[FIRST_Y_AXIS].link_udf->at) {
 	    fprintf(gpoutfile, "gnuplot.y2_mapping = function(y) { return y; };");
 	    fprintf(gpoutfile, "  // replace returned value with %s\n",
 			axis_array[FIRST_Y_AXIS].link_udf->definition);
